src/axom/inlet/tests/inlet_object.cpp

```diff
@@ -436,6 +436,40 @@ TYPED_TEST(inlet_object, implicit_conversion_primitives)
   EXPECT_EQ(arr, expected_arr);
 }
 
+TYPED_TEST(inlet_object, primitive_arrays_as_std_vector)
+{
+  std::string testString = " arr = { [0] = 4, [1] = 6, [2] = 10}";
+  DataStore ds;
+  Inlet inlet = createBasicInlet<TypeParam>(&ds, testString);
+
+  // Define schema
+  inlet.addIntArray("arr");
+
+  // Attempt both construction and assignment
+  std::vector<int> expected_arr {4, 6, 10};
+  std::vector<int> arr = inlet["arr"];
+  EXPECT_EQ(arr, expected_arr);
+  arr = inlet["arr"];
```

Comment:
Why is this here a second time? Does this test a separate path than the line above?

Reply:
@estebanpauli -- Despite the syntax, Line 450 actually uses a copy constructor, rather than the assignment operator.

Reply:
Copy construction vs. copy assignment is a slightly different overload resolution context that has caused problems with these implicit conversions in the past, so this test checks both to avoid regressions.

Reply:
Sorry, I meant a separate path for the conversion from the result of inlet["arr"] (a Proxy, I believe), to something that the vector can be constructed/assigned from. I guess it is important to test both though, since as you point out, they are not the same.